sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -391,6 +391,17 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
 
           case _ =>
         })
+
+        // Check for unresolved TABLE arguments after the main check above to allow other analysis
+        // errors to apply first, providing better error messages.
+        getAllExpressions(operator).foreach(_.foreachUp {
```

Comment:
Can we check this together with ` getAllExpressions(operator).foreach(_.foreachUp {` in line 290 so we don't need to iterate the plan again?

Reply:
No, because we need to check all the expressions for "unresolved attribute" or other errors first, before we do this check for `FunctionTableSubqueryArgumentExpression` instances.

That code on L290 starts with this:

```
        getAllExpressions(operator).foreach(_.foreachUp {
          case a: Attribute if !a.resolved =>
            failUnresolvedAttribute(operator, a, "UNRESOLVED_COLUMN")
        ...
```

We need to make sure we perform that "unresolved attribute" check for *all* expressions in the plan *before* we proceed further to check for `FunctionTableSubqueryArgumentExpression`.

Reply:
Ah I get it now. This is a node that should have been rewritten during analysis. cc @cloud-fan do you know if there are better places to add this?